### PR TITLE
Remove deprecated methods for AirWallMaterial, Node, SizingSystem, ZoneAirMassFlowConservation

### LINKED ISSUE #4063: Remove the deprecated AirWallMaterial class
There are two separate construction for air wall which when used gives different results for lighting load, 

As you can see in the screenshot I have assumed the Lighting density to be 1 W/ft2 for atrium, Using both of them gives different results, It looks the air wall material taken form library gives accurate result as the summation of the lighting loads results to 1W/ft2 unlike the construction material which gives more than 4W/ft2. Not sure about the radiance though.

## Current Behavior
Different Lighting load while using different set of air wall available.

## Expected Behavior
There should be only 1 air wall and it should  be the one that is available in library as air boundary construction as it gives more options like zone grouping and air mixing.

## Steps to Reproduce
1.Make an OSM model.
2.Assign a lighting density to a space (multifloor)
3.Use air wall from construction for ceiling and floor.
4.Use air boundary from library  and compare the results

## Possible Solution
Have only 1 air wall construction.

## Details
Windows, 10 Open studio app 1,0,1, 3.0.1

airwall

airwall1

atrium lighting

lights library airwall

library airwall

### LINKED ISSUE #4618: Remove functions in Node that have been deprecated for 9+ years
Issue overview
--------------

https://github.com/NREL/OpenStudio/blob/232fa2babef311ea8810a03b7abf170ca16ac433/src/model/Node.cpp#L244-L250

Do a blame, you'll see it's been there since the first public commit ever: https://github.com/NREL/OpenStudio/blame/232fa2babef311ea8810a03b7abf170ca16ac433/src/model/Node.cpp#L148

I didn't realize they were still there because they aren't marked `OS_DEPRECATED` in the public API.

## Current Behavior

They exist.

## Expected Behavior

We should delete them

### Environment

Some additional details about your environment for this issue (if relevant):
 - Platform (Operating system, version): all
 - Version of OpenStudio (if using an intermediate build, include SHA): 3.4.1-alpha

### Context

I went into the file as part of implementing https://github.com/NREL/OpenStudio/issues/4543

### PR DESCRIPTION
**Pull request by joseph-robertson**
Pull request overview
---------------------

 - Fixes #4618
 - Fixes #4063

Which deprecated methods should be removed?
- [ ] AirConditionerVariableRefrigerantFlow
- [ ] AirLoopHVAC
- [ ] AirTerminalSingleDuctVAVReheat
- [x] AirWallMaterial
  - Remove class completely; implement VT to ConstructionAirBoundary
- [ ] AvailabilityManagerNightCycle
- [ ] BuildingUnit
- [x] CoilCoolingDXSingleSpeed
  - https://github.com/NREL/OpenStudio/pull/4644
- [x] CoilCoolingDXTwoSpeed
  - https://github.com/NREL/OpenStudio/pull/4644 
- [ ] CoilCoolingWater
- [ ] CoilCoolingWaterToAirHeatPumpEquationFit
- [ ] CoilHeatingGas
- [ ] CoilHeatingWaterToAirHeatPumpEquationFit
- [ ] ConstructionAirBoundary
- [ ] CurveDoubleExponentialDecay
- [ ] DesignDay
- [ ] DesignSpecificationOutdoorAir
- [ ] EvaporativeCoolerDirectResearchSpecial
- [ ] ExteriorFuelEquipment
- [ ] ExteriorLights
- [ ] FanConstantVolume
- [ ] FanZoneExhaust
- [ ] GenericModelObject
- [ ] HeatPumpWaterToWaterEquationFitCooling
- [ ] HeatPumpWaterToWaterEquationFitHeating
- [ ] LayeredConstruction
- [x] Node (methods deprecated for 9+ years)
- [ ] ParentObject
- [ ] PlanarSurface
- [ ] PlantLoop
- [ ] PumpVariableSpeed
- [ ] ScheduleFile
- [ ] SiteWaterMainsTemperature
- [x] SizingSystem (note about removing after openstudio-standards 0.29 is released)
  - Remove `minimumSystemAirFlowRatio` in favor of `centralHeatingMaximumSystemAirFlowRatio`
  - Remove `setMinimumSystemAirFlowRatio` in favor of `setCentralHeatingMaximumSystemAirFlowRatio`
- [ ] SpaceInfiltrationDesignFlowRate
- [ ] StandardGlazing
- [ ] SubSurface
- [ ] ThermalZone
- [ ] ThermostatSetpointDualSetpoint
- [x] ZoneAirMassFlowConservation (methods deprecated as of OS 1.9.3 that do nothing); also fix RT
- [ ] ZoneHVACEquipmentList
- [ ] ZoneHVACLowTempRadiantConstFlow
- [ ] ZoneHVACPackagedTerminalAirConditioner
- [ ] ZoneHVACPackagedTerminalHeatPump
- [ ] ZoneHVACTerminalUnitVariableRefrigerantFlow
- [ ] ZoneVentilationDesignFlowRate

### Pull Request Author

 - [x] Model API Changes / Additions
 - [ ] Any new or modified fields have been implemented in the EnergyPlus ForwardTranslator (and ReverseTranslator as appropriate)
 - [ ] Model API methods are tested (in `src/model/test`)
 - [ ] EnergyPlus ForwardTranslator Tests (in `src/energyplus/Test`)
 - [ ] If a new object or method, added a test in NREL/OpenStudio-resources: Add Link
 - [ ] If needed, added VersionTranslation rules for the objects (`src/osversion/VersionTranslator.cpp`)
 - [ ] Verified that C# bindings built fine on Windows, partial classes used as needed, etc.
 - [ ] All new and existing tests passes
 - [ ] If methods have been deprecated, update rest of code to use the new methods

**Labels:**

 - [ ] If change to an IDD file, add the label `IDDChange`
 - [x] If breaking existing API, add the label `APIChange`
 - [x] If deemed ready, add label `Pull Request - Ready for CI` so that CI builds your PR

### Review Checklist

This will not be exhaustively relevant to every PR.
 - [ ] Perform a Code Review on GitHub
 - [ ] Code Style, strip trailing whitespace, etc.
 - [ ] All related changes have been implemented: model changes, model tests, FT changes, FT tests, VersionTranslation, OS App
 - [ ] Labeling is ok
 - [ ] If defect, verify by running develop branch and reproducing defect, then running PR and reproducing fix
 - [ ] If feature, test running new feature, try creative ways to break it
 - [ ] CI status: all green or justified

### CODE REVIEW

**Comment by tijcolem** [comment]
I think we should create a wiki page that describes the policy for deprecated methods. Do we support 2 versions? 1 Version? Use OS_DEPRECATE.  I think the warning should ouput the last supported version.

**Comment by joseph-robertson** [comment]
@jmarrec If we remove the `AirWallMaterial` class, do we need to create a VT from `Material:AirWall` to `Construction:AirBoundary`? Would it follow the now obsolete code from FT:
```
    // replace old style air wall Constructions with ConstructionAirBoundary
    for (LayeredConstruction construction : model.getModelObjects<LayeredConstruction>()) {
      if (construction.isModelPartition() && (construction.numLayers() == 1)) {
        MaterialVector layers = construction.layers();
        OS_ASSERT(layers.size() == 1u);

        // if this is an old style air wall
        if (layers[0].optionalCast<AirWallMaterial>()) {
          ConstructionAirBoundary newConstruction(model);
          newConstruction.setName(construction.nameString() + "_ConstructionAirBoundary");

          for (WorkspaceObject source : construction.sources()) {
            for (unsigned index : source.getSourceIndices(construction.handle())) {
              bool test = source.setPointer(index, newConstruction.handle());
              OS_ASSERT(test);
            }
          }

          LOG(Warn, "Construction '" << construction.nameString() << "' has been converted to ConstructionAirBoundary '"
                                     << newConstruction.nameString() << "'.");
          construction.remove();
        }
      }
    }
```

**Comment by jmarrec** [comment]
> @jmarrec If we remove the AirWallMaterial class, do we need to create a VT from Material:AirWall to Construction:AirBoundary? Would it follow the now obsolete code from FT:

Yes, absolutely. Scan constructions, find those that have only one layers that is an Material:AirWall, create a new Construction:AirBoundary, delete/don't VT the original construction and layer.

File: src/model/PlanarSurface.cpp [hunk 1]
```diff
@@ -228,10 +226,6 @@ namespace model {
           if (construction->numLayers() == 1) {
             MaterialVector layers = construction->layers();
             OS_ASSERT(layers.size() == 1u);
-            if (layers[0].optionalCast<AirWallMaterial>()) {
-              result = true;
-              LOG(Warn, "AirWallMaterial is deprecated, use ConstructionAirBoundary instead.");
-            }
```
Comment (jmarrec) [inline src/model/PlanarSurface.cpp:234]: Remove this entire block. Also need to update comment in hpp

File: src/model/SizingSystem.cpp [hunk 2]
```diff
@@ -1632,15 +1632,6 @@ namespace model {
...
   }
 
-  // DEPRECATED: TODO REMOVE as soon as standards > 0.29.0 is relased
-  boost::optional<double> SizingSystem::minimumSystemAirFlowRatio() const {
-    return getImpl<detail::SizingSystem_Impl>()->centralHeatingMaximumSystemAirFlowRatio();
-  }
-  // DEPRECATED: TODO REMOVE as soon as standards > 0.29.0 is relased
-  bool SizingSystem::setMinimumSystemAirFlowRatio(double centralHeatingMaximumSystemAirFlowRatio) {
-    return getImpl<detail::SizingSystem_Impl>()->setCentralHeatingMaximumSystemAirFlowRatio(centralHeatingMaximumSystemAirFlowRatio);
-  }
```
Comment (jmarrec) [inline src/model/SizingSystem.cpp:1642]: OpenStudio-resources needs adjustment: I'd want a PR that fixes stuff and confirmation that everything still runs.

this one is disabled but should still be fixed: https://github.com/NREL/OpenStudio-resources/blob/6c90661c95c0e896ac447eefaa153c9205319534/model/simulationtests/afn_single_zone_ac.rb#L56

File: src/model/ZoneAirMassFlowConservation.cpp [hunk 3]
```diff
@@ -115,12 +107,6 @@ namespace model {
       OS_ASSERT(result);
     }
 
-    bool ZoneAirMassFlowConservation_Impl::setSourceZoneInfiltrationTreatment(const std::string& sourceZoneInfiltrationTreatment) {
-      return false;
-    }
```
Comment (jmarrec) [inline src/model/ZoneAirMassFlowConservation.cpp:120]: https://github.com/NREL/OpenStudio-resources/blob/6c90661c95c0e896ac447eefaa153c9205319534/model/simulationtests/zone_mixing.rb#L71

**Review by jmarrec — CHANGES_REQUESTED** [review]

File: src/osversion/VersionTranslator.cpp [hunk 4]
```diff
@@ -6911,9 +6913,73 @@ namespace osversion {
...
+      if (iddname == "OS:Construction") {
+
+        // Remove Construction with Material:AirWall layer
+        // Replace with Construction:AirBoundary
+
+        auto iddObject = idd_3_4_1.getObject(iddname);
+
+        bool isAirWall = false;
+        if (object.numExtensibleGroups() == 1u) {
+          for (const IdfExtensibleGroup& eg : object.extensibleGroups()) {
```
Comment (jmarrec) [inline src/osversion/VersionTranslator.cpp:6936]: for loop useless if you have 1 extensible group.

**Comment by jmarrec** [comment]
I fixed the regression tests in https://github.com/NREL/OpenStudio-resources/pull/174

File: src/osversion/VersionTranslator.cpp (hunk 4 above)
Comment (jmarrec) [reply src/osversion/VersionTranslator.cpp]: done in 51e36b5220938c78707ef1a4d6d00b1242463b49

**Comment by jmarrec** [comment]
@joseph-robertson I'll let this one build. Then I'll do another run of https://github.com/NREL/OpenStudio-resources/pull/174 to make sure everything is alright and then we can merge. Thanks for your work!

**Pull request merged**